bluesky/callbacks/scientific.py

```diff
@@ -167,8 +167,8 @@ def compute(self):
         if _cen_list:
             self.cen = np.mean(_cen_list)
 
-        if len(_cen_list) == 2:
-            self.fwhm = float(_cen_list[1] - _cen_list[0])
+        if len(_cen_list) >= 2:
```

Comment:
For noisy data, we will get cen_list lengh is larger than 2. If we define it equal to 2, we get None as return on fwhm. 

This is consistent with chx algo now. https://github.com/NSLS-II-CHX/ipython_ophyd/blob/d2389af73506d6cf2f70e95a336b77a16267381f/profile_collection/startup/95-utilities.py#L275

Reply:
Good catch! I notice that CHX's algo takes the absolute value. Would that be appropriate here as well?

Reply:
yeah good catch. maybe it's also worth doing the same for `self.cen`. if `_cen_list > 2`, it might not make sense anymore.
I'd be tempted (but this is probably overkill, also writing this naively.):
```python
_cen_list = _cen_list.sort()
self.cens = (_cen_list[1:] + _cen_list[:-1])*.5
self.fwhms = np.diff(_cen_list)
# choose the cen or peak, but allow user to look at the others? 
self.cen = self.cens[0]
self.fwhms = self.fwhm[0]
```

Reply:
Good suggestion. I think this may work for late update when we want to deal with multiple peaks. 

But for single peak with lots of noise, the middle curve cuts multiple points on both the left and right of the peak. That's what we are focusing now. And this will fail I think.

Reply:
ok sounds good. i agree. this PR should be restricted to the `self.fwhm=None` issue. maybe a PR should be made following this one?
Great tests!!